### ISSUE TITLE
Option to auto-play

### ISSUE THREAD

**Issue by gingerbeardman** [benwiggy/APPlayMIDI#6]
I'd like to set it so that it auto-played whatever MIDI file i double clicked to open

Thanks!

**Comment by benwiggy** [comment]
Nice idea. That'll probably have to go alongside ensuring that only one file is playing at any one time, to avoid chaos! And need a preference for doing so, or not.

**Comment by gingerbeardman** [comment]
FWIW QuickTime Player 7 does it (I'm still on Mojave so can still just about run that)

**Comment by shytzedaka** [comment]
Yeah this is a great idea! really loving the app

**Comment by Organist22** [comment]
+1

**Comment by abulka** [comment]
Autoplay is much needed by me, as well as a loop option, and an option to select drums ch 10.

**Comment by benwiggy** [comment]
The app simply passes the MIDI data to the OS's built-in MIDI player, so I can't do anything about assigning instruments. 
I'll have another look at auto-play and looping.

**Comment by leadovation-tech** [comment]
Hello. Did you plan to add this autoplay feature ? Definitely interested by this 😉

**Comment by benwiggy** [comment]
Yes, it's definitely something I'd like to include. But as stated above, I would need to make sure that  when a whole bunch of files are  opened together, they don't all play. And I dare say that some people might not want auto-play; which would involve create a preferences dialog and controls.

So, as ever, simple things are not simple.

**Comment by abulka** [comment]
If one file is selected, then auto play.  If multiple files are selected then auto play them in sequence, or if thats too hard, just autoplay the first one. When you navigate to the next file within the preview tool window, stop the previous and play the next. No need for settings.

Hitting space bar to preview any file type on Mac results in seeing or hearing immediately - that's the Mac standard. Having to click a play button is an extra annoying step that is not required by any other Mac preview tool that I know of - my feeling is that everyone wants and expects auto play behaviour, so there is no need for settings to configure this behaviour.

**Comment by benwiggy** [comment]
The trouble is that QuickLook _does_ require users to click on a play button in the Preview panel of a Finder window. No one would want those files to start playing immediately just because you selected them. Obviously, pressing Space bar to bring up a QL window is itself a user interaction, and video/audio content starts immediately.

However, I've yet to find any documentation for how to differentiate between QuickLook in the Finder Preview panel and QuickLook in a new window. 

I'll have a look at trying to get some sort of Document manager that oversees how many windows there, and which one is playing, etc.
If anyone has any ideas, all forks welcome!
